### DISCUSSION TITLE
Compiling a specific file

### DISCUSSION THREAD

**Discussion by Tului2020** [nrwl/nx#17202] [General]
Hello, 

Project Details: ExpressJS/NodeJS Backend

I am working on a multithreading project and in order for it to work, it needs a separate file that is compiled to JS, right now I only have main.js. 

Is there a way to configure the project.json file to pick up a file (.ts) in the "assets" directory, and replace it with a compiled version (.js) of it? If not, have people done projects like this?

Thanks!

**Comment by ImTemporaryHere** [comment]
"options": {
       ...
        "additionalEntryPoints": [
          {
            "entryName": "worker",
            "entryPath": "path/worker.ts"
          }
        ]
      },

> Hello,
> 
> Project Details: ExpressJS/NodeJS Backend
> 
> I am working on a multithreading project and in order for it to work, it needs a separate file that is compiled to JS, right now I only have main.js.
> 
> Is there a way to configure the project.json file to pick up a file (.ts) in the "assets" directory, and replace it with a compiled version (.js) of it? If not, have people done projects like this?
> 
> Thanks!

  "options": {
       ...
        "additionalEntryPoints": [
          {
            "entryName": "worker",
            "entryPath": "path/worker.ts"
          }
        ]
      },
